### PR TITLE
News 15

### PR DESCRIPTION
**Pull request by RootDevil**
- Add GET /api/articles/:article_id/comments endpoint
- Test status 200 code with array of comments
- Test status 200 code with empty array
- Test 404 and 400 error codes
- Add `checkResourceExists()` util function

### CODE REVIEW

File: models/comments.model.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+exports.selectCommentsByArticleId = async (articleId) => {
+    await checkResourceExists('articles', 'article_id', articleId);
+
+    const comments = await db.query(`
+        SELECT * FROM comments
+        JOIN articles ON comments.article_id = articles.article_id
+        WHERE articles.article_id = $1;
+    `, [articleId]);
+
```
Comment (emilyb93) [inline models/comments.model.js:12]: Do we need to join articles to this table? The comments data should already have the article_id on them

File: utils/utils.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const db = require('../db/connection');
+const format = require('pg-format');
+
+exports.checkResourceExists = async (table, column, value) => {
+    const queryStr = format('SELECT * FROM %I WHERE %I = $1;', table, column);
+    const queryOutput = await db.query(queryStr, [value]);
```
Comment (emilyb93) [inline utils/utils.js:6]: Nothing to say, just a nice use of pgformat here

**Review by emilyb93 — COMMENTED** [review]
Just have a look at the query on selectCommentsByArticleId, can we achieve this without a Join?

**Comment by RootDevil** [comment]
- Removed JOIN statement

**Review by emilyb93 — APPROVED** [review]
Brill looks good

**Pull request merged**